src/azure-cli/azure/cli/command_modules/acr/manifest.py

```diff
@@ -0,0 +1,454 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+try:
+    from urllib.parse import unquote
+except ImportError:
+    from urllib import unquote
+
+from knack.log import get_logger
+from azure.cli.core.util import user_confirmation
+from azure.cli.core.azclierror import InvalidArgumentValueError
+
+from .repository import (
+    _parse_image_name,
+    get_image_digest,
+    _obtain_data_from_registry,
+    _get_manifest_path,
+    _acr_repository_attributes_helper
+)
+
+from ._docker_utils import (
+    request_data_from_registry,
+    get_access_credentials,
+    get_login_server_suffix,
+    RepoAccessTokenPermission
+)
+
+from ._validators import (
+    BAD_MANIFEST_FQDN,
+    BAD_REPO_FQDN
+)
+
+logger = get_logger(__name__)
+
+ORDERBY_PARAMS = {
+    'time_asc': 'timeasc',
+    'time_desc': 'timedesc'
+}
+DEFAULT_PAGINATION = 100
+
+BAD_ARGS_ERROR_REPO = "You must provide either a fully qualified repository specifier such as"\
+                      " 'MyRegistry.azurecr.io/hello-world' as a positional parameter or"\
+                      " provide '-r MyRegistry -n hello-world' argument values."
+BAD_ARGS_ERROR_MANIFEST = "You must provide either a fully qualified manifest specifier such as"\
+                          " 'MyRegistry.azurecr.io/hello-world:latest' as a positional parameter or provide"\
+                          " '-r MyRegistry -n hello-world:latest' argument values."
+
+
+def _get_v2_manifest_path(repository, manifest):
+    return '/v2/{}/manifests/{}'.format(repository, manifest)
+
+
+def _get_referrers_path(repository, manifest):
+    return '/oras/artifacts/v1/{}/manifests/{}/referrers'.format(repository, manifest)
+
+
+def _obtain_manifest_from_registry(login_server,
+                                   path,
+                                   username,
+                                   password,
+                                   raw=False):
+
+    result, _ = request_data_from_registry(http_method='get',
+                                           login_server=login_server,
+                                           path=path,
+                                           raw=raw,
+                                           username=username,
+                                           password=password,
+                                           result_index=None,
+                                           manifest_headers=True)
+
+    return result
```

Comment:
usually, we define the API calls in one of the helpers/handlers files so we can call them. This isn't wrong just make sure this method being here are the best option and follow the current code style

Reply:
I would recommend make _poller file to put everything that handles the API calls and have in this file the methods that work for the command group only